lib/fluent/plugin/in_tail.rb

```diff
@@ -389,7 +389,7 @@ def setup_watcher(target_info, pe)
       end
 
       if @enable_stat_watcher
-        tt = StatWatcher.new(path, log) { tw.on_notify }
+        tt = StatWatcher.new(target_info.path, log) { tw.on_notify }
```

Comment:
Oh my god....

Reply:
I got it why tests aren't failed even for such broken code.
When there is no wildcard in the path, `path` is valid because it's defined as config_param:
```
    config_param :path, :string
```
and `path` and `target_info.path` are same.

On the other hand, when a wildcard is used in the `path` config, the expanded path (`target_info.path`) and the `path` param aren't matched. 
e.g.)

* `path` param: tail*.txt
* `target_info.path`: tail1.txt

In addition, there is no test for the later case.

Reply:
:clap: so impressed with this find!